flomio/passninja-guestlist-gas
# Refactor virtualization: fixes #39, fixes #42

# Changelog
- Created an OOP virtual mimic of Google Spreadsheet akin to the Shadow Dom to allow more efficient read/writes to/from the spreadsheets, effectively caching until flush
- Added test suite and ability to do testing in the first place.
- Overhauled npm run scripts to be easier to use
- Applied linting to gs files for vscode and removed the necessity to convert the files (other than to make documentation).  
- Implemented the virtualization across all necessary functions (ones that do requests/time sensitive).
- Augmented logging to have time taken in ms and similar structure to npm debug module + basic filtering.
- Removed unused functions + moved catchError to errors.

# To Verify
- Run all functions you can think of in a live sheet (the build scripts are unaffected) and ensure they all still work + submit from a live form.
